tests/unit/test_logger.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+import uuid
+from typing import Generator, Optional
+
+import pytest
+
+from dss.logger import setup_logger  # Adjust with your actual module name
+
+
+@pytest.fixture(scope="function")
+def logger_setup(
+    tmp_path_factory: pytest.TempPathFactory,
+) -> Generator[logging.Logger, None, None]:
+    """Fixture to set up a unique logger with a log file for the test."""
+    # Generate a random and unique logger name to avoid conflicts
+    unique_logger_name = f"test_logger_{uuid.uuid4().hex}"
+
+    # Set a static log file name, unique to each test
+    log_file_name = "test_log.log"
+    log_file_path = tmp_path_factory.mktemp("logs") / log_file_name
+
+    # Create a unique logger instance
+    logger = logging.getLogger(unique_logger_name)
+
+    # Clear any existing handlers (if reusing logger names between tests)
+    logger.handlers.clear()
+
+    logger = setup_logger(
+        log_file_path=str(log_file_path),
+    )
+
+    # Force creation of the log file by logging an empty message
+    logger.debug("Logger initialized")
+
+    yield logger
+
+    # Cleanup: remove the log file after the test is done
+    if log_file_path.exists():
+        os.remove(log_file_path)
+
+
+def get_log_file_path(logger: logging.Logger) -> Optional[str]:
+    """Helper function to get the file path of the log file used by the logger."""
+    for handler in logger.handlers:
+        if isinstance(handler, logging.FileHandler):
+            return handler.baseFilename
+    return None
+
+
+@pytest.mark.parametrize(
+    "log_level, log_message, expected_prefix",
+    [
+        (logging.INFO, "This is an info message", "[INFO]"),
+        (logging.WARNING, "This is a warning message", "[WARNING]"),
+        (logging.DEBUG, "This is a debug message", "[DEBUG]"),
+        (logging.ERROR, "This is an error message", "[ERROR]"),
+    ],
```

Comment:
Refactor here to have the logging levels in increasing numeric value
```suggestion
    [
        (logging.DEBUG, "This is a debug message", "[DEBUG]"),
        (logging.INFO, "This is an info message", "[INFO]"),
        (logging.WARNING, "This is a warning message", "[WARNING]"),
        (logging.ERROR, "This is an error message", "[ERROR]"),
    ],
```